Source/Core/DolphinQt/RenderWidget.cpp

```diff
@@ -530,11 +530,10 @@ void RenderWidget::PassEventToPresenter(const QEvent* event)
     const u32 key = static_cast<u32>(key_event->key() & 0x1FF);
 
     const char* chars = nullptr;
-    QByteArray utf8;
 
     if (is_down)
     {
-      utf8 = key_event->text().toUtf8();
+      QByteArray utf8 = key_event->text().toUtf8();
```

Comment:
This is an object lifetime issue. With this change, `const char* chars` will still contain the result of `utf8.constData()` after `QByteArray utf8` goes out of scope. This is why the code was written the way it was before. [This is not the first time](https://github.com/dolphin-emu/dolphin/pull/13025#pullrequestreview-2245509993) your automated changes have produced erroneous code like this.